subsys/bootloader/bl_storage/nrf_nv_counters.c

```diff
@@ -56,14 +56,15 @@ int32_t boot_nv_security_counter_update(uint32_t image_id, uint32_t img_security
 	 */
 	fih_int security_cnt;
 
-	fih_int fih_err = boot_nv_security_counter_get(image_id, &security_cnt);
+	fih_int fih_err;
 
-	if (fih_err != FIH_SUCCESS) {
-		return fih_err;
+	FIH_CALL(boot_nv_security_counter_get, fih_err, image_id, &security_cnt);
+	if (FIH_NOT_EQ(fih_err, FIH_SUCCESS)) {
+		return -1;
 	}
 
 	if (security_cnt == (uint16_t)img_security_cnt) {
```

Comment:
incorrect handling of fih_int I believe. (fih_int can be compiled to be a struct, so need to use some of their magic fih-macros or functions to compare)

Reply:
Probably only issue in HIGH_PROFILE, not sure if we support or that, or want to have code that's broken for that case anyways

Reply:
I see no  reason to not use the macros I think. Even if we don't  support it (so we don't test it) it's a better practice to use the macros that they provide to us I think.